src/kokkos_abstraction.hpp

```diff
@@ -635,6 +635,36 @@ struct SpaceInstance<Kokkos::Cuda> {
 };
 #endif
 
+struct DeviceDeleter {
```

Comment:
You probably should make the execution space a member of the deleter since at some point we intend to do more stuff with streams.

Reply:
`DevExecSpace` is a type, I think, so I'm not sure how to store this -- would it help to template `DeviceDeleter` on `ExecSpace` and `MemSpace`?

Reply:
Yeah, that's right.

Reply:
I'm not familiar with the stream stuff but I added the template parameters to `DeviceAllocate` and `DeviceCopy` on the assumption that they will be needed there as well.

Reply:
I'm not familiar with the stream stuff but I added the template parameters to DeviceAllocate and DeviceCopy on the assumption that they will be needed there as well.

Reply:
Oh, you don't need the `ES` anymore if you're not calling the destructor.

Reply:
nice catch, thanks (fixed)